bin/authentication.py

```diff
@@ -59,10 +88,20 @@ def check_password(environ, user, password):
         engine, connection, metadata = open_database()
         authenticated = check_database(user, password, connection, metadata)
 
-    if authenticated is not True or user == user_id:
+    if authenticated is not True or (not is_team and user == user_id):
```

Comment:
If this is a team assignment, is there a case where the "user_id" that's gotten from the path is going to correspond to a user that isn't on the team?